hack/xref-helpmsgs-manpages

```diff
@@ -35,6 +35,43 @@ my $Markdown_Path = "$Docs_Path/markdown";
 # Global error count
 my $Errs = 0;
 
+# Table of exceptions for documenting fields in '--format {{.Foo}}'
+#
+# Autocomplete is wonderful, and it's even better when we document the
+# existing options. Unfortunately, sometimes internal structures get
+# exposed that are of no use to anyone and cannot be guaranteed. Avoid
+# documenting those. This table lists those exceptions. Format is:
+#
+#      foo       .Bar
+#
+# ...such that "podman foo --format '{{.Bar}}'" will not be documented.
+#
+my $Format_Exceptions = <<'END_EXCEPTIONS';
+# Deep internal structs; pretty sure these are permanent exceptions
+events       .Details
+history      .ImageHistoryLayer
+images       .ImageSummary
+network-ls   .Network
+
+# FIXME: this one, maybe? But someone needs to write the text
+machine-list    .Starting
+
+# No clue what these are. Some are just different-case dups of others.
+pod-ps  .Containers .Id .InfraId .ListPodsReport .Namespace
+ps      .Cgroup .CGROUPNS .IPC .ListContainer .MNT .Namespaces .NET .PIDNS .User .USERNS .UTS
+
+# I think .Destination is an internal struct, but .IsMachine maybe needs doc?
+system-connection-list .Destination .IsMachine
+END_EXCEPTIONS
```

Comment:
@Luap99 PTAL, this is the most questionable part of my PR

Reply:
I think this is a reasonable way of starting this. However I see two problem in this:
 - Every new unwanted field must be special cases here in the script. Not something most authors will be aware of.
 - The docs will not match the completion. If we do not want some stuff documented should we maybe not auto complete them as well? I think having a solution in *pure* go would be better, something that allows us to set a option on a struct field to make it not show up. We could also write some form of generator that will read the go comments from the struct fields and use this as Description so that a) a human must not write them by hand and b) code changes will be reflected in the docs. Of course the quality of the go comments varies but I think most are decent enough to be used as user doc.
That said I know that is not something you can write and I don't know how difficult it would be for me or somebody else.

I am definitely in favour of merging this though. This is much better than nothing.

Reply:
>Every new unwanted field must be special cases here in the script. Not something most authors will be aware of.

True. Good point.

>If we do not want some stuff documented should we maybe not auto complete them as well? 

Exactly, and this is what has blocked me the two other times I've tried to submit something like this: the document-or-not argument has been unsolvable. See https://github.com/containers/podman/pull/14386#discussion_r882991797

I would *love* for those tables to be autogenerated!